data_structures/hash_table/hash_table.py

```diff
@@ -0,0 +1,81 @@
+from linked_list import LinkedList
+
+
+class HashTable:
+    def __init__(self, max_size=1024):
+        """
+        This initiatates the hash table with empty linked lists in each bucket.
+        """
+
+        if type(max_size) is not int:
+            raise TypeError
+
+        linked_list = LinkedList()
+        self.max_size = max_size
+        self.buckets = [linked_list] * self.max_size
+
+    def hash_key(self, key):
+        """
+        assigns a int value to key
+        """
+        if type(key) is not str:
+            raise TypeError
+
+        sum = 0
+        for char in key:
+            sum += ord(char)
+        return sum % len(self.buckets)
+
+    def set(self, key, val):
+        """
+        inserts {key: val} into hash table.
+        """
+
+        if type(key) is not str:
+            raise TypeError
+
+        return self.buckets[self.hash_key(key)].insert({key: val})    
+
+        # hash the key; get a location for the bucket to insert into
+        # set val into bucket
+
+    def get(self, key):
+        """
+        uses key to return value from index.
+        """
+
+        if type(key) is not str:
+            raise TypeError
+
+        node = self.buckets[self.hash_key(key)].head
+
+        while node:
+            if key in node.val.keys():
```

Comment:
To check if a key is in a dictionary, you can just check if `key in node.val`, you do no need to look in `.keys()` .